src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategyTransformer.kt

```diff
@@ -950,23 +950,24 @@ internal class ManagedStrategyTransformer(
     }
 
     /**
-     * Removes switch points in CancellableContinuationImpl.cancel, so that they will not be reported
-     * when a continuation is cancelled by lincheck
+     * Removes ignored section for `CancellableContinuationImpl.onCancellation` call, so that
+     * events in the `onCancellation` method influence the execution and the trace.
+     *
+     * Note that this transformer does not handle the case of cancel handler throwing an exception.
      */
     private inner class CancellabilitySupportMethodTransformer(methodName: String, mv: GeneratorAdapter) : ManagedStrategyMethodVisitor(methodName, mv) {
```

Comment:
Can we remove this transformer and add `invokeAfterIgnoredSectionLeaving` and `invokeBeforeIgnoredSectionEntering` invocations to the runner's `cancelByLincheck` implementation?